### ISSUE TITLE
SmingRTOS with ESP-IDF 1.0

### ISSUE THREAD

**Issue by jobidon** [SmingHub/SmingRTOS#151]
I am about to start development for the ESP32 and am evaluating the options available. I am looking for a framework (like sming) with low level access to the h/w, but that will support the newer SDK from ESP. And, according to a quick evaluation, the other solutions (lua, js and even arduino), seem to abstract the hardware too much.

I understand that @hreintke is no longer supporting the sming framework, but is anyone still using it? Specifically, has anyone tried it with the latest version of the ESP-IDF?

**Comment by salocinx** [comment]
I am currently try to find out the same. Can I use SmingRTOS on the ESP-32 with ESP-IDF ?

**Comment by jobidon** [comment]
I suggest you stick with ESP-IDF: smingRTOS is no longer supported

**Comment by salocinx** [comment]
@jobidon: Thanks for answering! Nevertheless I don't give up the hope yet, since my goal is straightforward. Would be great if you could give a short feedback on the following:

I am trying to port the [MavBridge](https://github.com/festlv/mavbridge) project to ESP32, which itself is about 2 years old and according to the [README](https://github.com/festlv/mavbridge/blob/master/README.md) it was developed on the ESP8266 with the help of the [Sming library 2.1.0](https://github.com/SmingHub/Sming). The MavBridge project is rather small, but heavily depends on Sming.

When having a look at the [SmigRTOS library's README](https://github.com/SmingHub/SmingRTOS/blob/master/README.md), it states that most of the classes have been ported up to Sming release 2.1.0 too.

Basically, the MavBridge project makes use of the `HTTP`, `AJAX`, `UdpConnection` and some `simplified GIOP access` of the Sming library.

Do you think there's any chance for me to use the "latest" SmingRTOS 2.1.0 framework to port the MavBridge (depending on Sming 2.1.0) project to ESP32?

**Comment by hreintke** [comment]
@salocinx 
I have created and maintained the SmingRTOS project.

SmingRTOS depends on/is running on top of the Espressif RTOS SDK. 
Although RTOS SDK and ESP-IDF are both RTOS based they are not compatible.
Besides that, a port for ESp32 would require the rebuild/adaption of the low-level GPIO/Interrupt routines.

In short, I see no easy way to port/use SmingRTOS 2.1.0 on ESP32.

As I am doing for my projects for ESP32, it is wise to go for ArduinoESP32, with AsyncTCP/AsyncUDP/AsyncWebserver libraries which provide most of the functionality available within SmingRTOS.
Alternative, but would take more effort, is to write on ESP-IDF directly.

**Comment by salocinx** [comment]
@hreintke thank you for the clarification and your suggestion to use arduino-esp32 library. I will go this way, since writing the required functionality on ESP-IDF would currently ask too much for me. Thank you!
